packages/fxa-settings/src/pages/PostVerify/ThirdPartyAuthCallback/index.tsx

```diff
@@ -140,6 +104,54 @@ const ThirdPartyAuthCallback = ({
     [finishOAuthFlowHandler, integration, location.search]
   );
 
+  const verifyThirdPartyAuthResponse = useCallback(async () => {
+    const { code: thirdPartyOAuthCode, provider } = (
+      integration as ThirdPartyAuthCallbackIntegration
+    ).thirdPartyAuthParams();
+
+    if (!thirdPartyOAuthCode) {
+      return hardNavigate('/');
+    }
+
+    try {
+      const linkedAccount: LinkedAccountData =
+        await account.verifyAccountThirdParty(
+          thirdPartyOAuthCode,
+          provider,
+          undefined,
+          queryParamsToMetricsContext(
+            flowQueryParams as unknown as Record<string, string>
+          )
+        );
+      await storeLinkedAccountData(linkedAccount);
+
+      setCurrentAccount(linkedAccount.uid);
+
+      const fxaParams = (
+        integration as ThirdPartyAuthCallbackIntegration
+      ).getFxAParams();
+
+      if (fxaParams.includes('client_id')) {
```

Comment:
Noticed that this does not add the query params if integration is sync fxa_desktop_v3